### PR TITLE
Join eager relations of a specified relation

### LINKED ISSUE #1416: Loading a relation should load that relations eager relations
I have a case where there is a relationship between two entities (`User` and `Photo`). `Photo` meanwhile has an eager relationship with `PhotoMetadata`. If I run this query,  
```js
const user = userRepository.find({ where: { id: 1}, relations: ['photo'] })

// I should be able to access this
user.photo.photoMetadata
```

shouldn't the `Photo` that gets joined with `User` have `PhotoMetadata` eagerly loaded? Currently, the eager relations for the relation are ignored

### PR DESCRIPTION
**Pull request by BrainMaestro**
Fixes #1416

### CODE REVIEW

File: src/entity-manager/EntityManager.ts [hunk 1]
```diff
@@ -496,7 +495,14 @@ export class EntityManager {
         const metadata = this.connection.getMetadata(entityClass);
         const qb = this.createQueryBuilder(entityClass, FindOptionsUtils.extractFindManyOptionsAlias(optionsOrConditions) || metadata.name);
         FindOptionsUtils.applyFindManyOptionsOrConditionsToQueryBuilder(qb, optionsOrConditions);
-        this.joinEagerRelations(qb, qb.alias, metadata);
+
+        ids = ids.map(id => {
```
Comment (pleerock) [inline src/entity-manager/EntityManager.ts]: why do you need those changes? Looks like an old code from the `master` branch. Are you sure you did a proper branch from next and not from master?

**Comment by pleerock** [comment]
Tests are failing as well. I think because of code from the master.

File: src/entity-manager/EntityManager.ts (hunk 1 above)
Comment (BrainMaestro) [reply src/entity-manager/EntityManager.ts]: Ah, rebase gone wrong. I branched from master at first, then rebased with next. Fixed

**Comment by BrainMaestro** [comment]
All good now

**Comment by pleerock** [comment]
Thank you very much!

**Pull request merged**